### DISCUSSION TITLE
ODE Toolbox

### DISCUSSION THREAD

**Discussion by bilalshaikh42** [biosimulations/biosimulations#2774] [Ideas]
@jonrkarr Curious about your thoughts on this work and any applicability to ours. Seems like it might have some? relevance to  our thoughts on suggesting simulation algorithms?

https://ode-toolbox.readthedocs.io/en/master/

**Comment by jonrkarr** [comment]
Sounds like this builds on SciPy. GillesPy2 exposes most of those same integration methods, albeit without a tool that helps users choose the optimal integrator.

COPASI's LSODA/LSODAR implementation also has logic to select among multiple choices for nested algorithms.

However, SciPy doesn't have the most advanced integration methods. Seems like the consensus is that CVODE is frequently the best and fastest. This is what tellurium, VCell, AMICI, PySCeS, OpenCOR, NEURON, NetPyNe, and others use. Some Python users might choose other tools because there isn't a great lower level Python interface to CVODE.

CVODE is also an entrypoint to multiple nested methods. I'm not sure what CVODE does to automatically recommend choices for these nested methods.

**Comment by johnsekar** [comment]
I actually don't see the point of this package. Typically the question is
whether a model is stiff or not under the conditions examined, so it is
often immediately apparent when you run it at specific parameter settings.
Programs that wrap CVODE often just run it and fail and then suggest using
the stiff solver options. And theoretically, any model could be stiff under
arbitrary parameter values. Maybe I'm missing something here, but are there
really that many options for solving ODEs that there needs to be a package
for resolving them? Seems like between CVODE and LSODA it covers a wide
gamut of tools.

**Comment by hsauro** [comment]
Our experience with running over 1000 models with the scipy integrators is
that they aren't suitable for large or very stiff models. Instead, we ended
up creating our own CvODE Python interface which worked fine.

Herbert

On Fri, Aug 6, 2021 at 11:03 PM John Sekar ***@***.***> wrote:

> I actually don't see the point of this package. Typically the question is
> whether a model is stiff or not under the conditions examined, so it is
> often immediately apparent when you run it at specific parameter settings.
> Programs that wrap CVODE often just run it and fail and then suggest using
> the stiff solver options. And theoretically, any model could be stiff under
> arbitrary parameter values. Maybe I'm missing something here, but are there
> really that many options for solving ODEs that there needs to be a package
> for resolving them? Seems like between CVODE and LSODA it covers a wide
> gamut of tools.
>
>
>
> On Fri, Aug 6, 2021, 13:21 Jonathan Karr ***@***.***> wrote:
>
> > Sounds like this builds on SciPy. GillesPy2 exposes most of those same
> > integration methods, albeit without a tool that helps users choose the
> > optimal integrator.
> >
> > COPASI's LSODA/LSODAR implementation also has logic to select among
> > multiple choices for nested algorithms.
> >
> > However, SciPy doesn't have the most advanced integration methods. Seems
> > like the consensus is that CVODE is frequently the best and fastest. This
> > is what tellurium, VCell, AMICI, PySCeS, OpenCOR, NEURON, NetPyNe, and
> > others use. Some Python users might choose other tools because there
> isn't
> > a great lower level Python interface to CVODE.
> >
> > CVODE is also an entrypoint to multiple nested methods. I'm not sure what
> > CVODE does to automatically recommend choices for these nested methods.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/biosimulations/Biosimulations/discussions/2774#discussioncomment-1140371
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAMJHQK2CHM456SNM35JVLLT3QK3FANCNFSM5BWIDF4A
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> >
> > .
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/biosimulations/Biosimulations/discussions/2774#discussioncomment-1141688>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIBSDRTKG7P4CTRUGZOBCTT3TEDDANCNFSM5BWIDF4A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Herbert Sauro, Professor
University of Washington, Bioengineering
206-685-2119, www.sys-bio.org
***@***.***
Books: http://books.analogmachine.org/
